cmake/nunicode.cmake

```diff
@@ -0,0 +1,38 @@
+macro(mbgl_nunicode_core)
```

Comment:
🤔 Standalone static probably makes sense and is most similar to how we used to load it with mason. I'll look into setting that up (my CMake game is not that strong, but time to learn).

Do you have any concerns about the manual stripping of unused files I did in the process of vendoring? It's a potential source of confusion, but just helps reduce the total number of lines of code that have to get pulled into the repo, included in search, etc...

Reply:
The define flags actually have to be public because they toggle sections of the header files on and off (unless we just hard-wire the headers to the state we want).